_posts/2018-09-13-is-cockpit-secure.md

```diff
@@ -0,0 +1,257 @@
+only the javascript code installed on the system is run, and bugs are not
+exploited to run code trojaned into logs or other data loaded by Cockpit?
+
+Browsers have a security technology called
+[Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+which is sorta like SELinux or Apparmor in your browser. The policy describes
+exactly where code can be loaded, what can be run, and what sorts of connections
+can be made by the browser.
+
+In Cockpit's case we send a strict ```Content-Security-Policy``` header that
+only allows code installed in Cockpit packages on the logged into system to be run.
+Although individual parts of Cockpit can
+[override this default](http://cockpit-project.org/guide/latest/packages.html#package-manifest),
+it's rarely done. The default security policy looks like this:
+
+     Content-Security-Policy: default-src 'self' connect-src 'self' ws: wss:
+
+A failure of *Content Security Policy* will look something like this in your browser's
+javascript console:
+
+![Content Security Policy denial](/images/csp-denied.png)
+
+## Security of the network facing TCP port
+
+Cockpit typically listens on TCP port 9090 on a host. This is the
+[websm or "Web Systems Manager" network port](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt).
+Opening a network facing port has security risks. Both the exposed surface
+area and stuff listening on that port, in this case ```cockpit-ws```, are risky.
+
+If this is an issue for your systems, you can use Cockpit over the SSH port
+already have open. But wait, how does that work? Browsers don't natively speak
+the SSH protocol, and getting a browser to do so would be an impressive party trick.
+
+To use Cockpit over SSH, use a *bastion host model*: Start ```cockpit-ws``` on port 9090
+on a single host, perhaps even your localhost. Connect to that with your
+browser and use that first Cockpit instance to log into to Cockpit on other machines
+others over SSH. If you draw it up, it looks something like this:
+
+![Multi-host Transport](/images/transport-multi-small.png)
+
+[Atomic Host includes Cockpit by default](http://www.projectatomic.io/docs/cockpit/),
+in this way. Atomic Host doesn't include ```cockpit-ws``` or open port 9090 by default,
+but expects you to connect from another running Cockpit instance over port 22. There's
+also the possibility to run ```cockpit-ws``` as a container to accomplish this.
+
+When you're trying this out in real life, specify an alternate server on the Cockpit
+login screen. The SSH protocol will be used to connect to it:
+
+![Login Alternate Server](/images/cockpit-login-alternate.png)
+
+Or you can add other machines to a local dashboard, and Cockpit will connect to
+them via SSH. Even usage of SSH key based authentication works great:
+
+<iframe width="853" height="480" src="https://www.youtube.com/embed/Ye5YlVNXC9w" frameborder="0" allowfullscreen></iframe>
+
```

Comment:
The video is not rendered.

Reply:
Strange it looks just like all the other posts.

Reply:
I was looking at the rendered output here by clicking "View" at the top of the file. That probably just doesn't allow arbitrary html.